pkg/config/structure/unmarshal.go

```diff
@@ -0,0 +1,368 @@
+	}
+	if v.Kind() == reflect.Slice {
+		return &nodeimpl{typ: listNodeType, val: v}
+	}
+	if isScalar(v) {
+		return &nodeimpl{typ: scalarNodeType, val: v}
+	}
+	panic(fmt.Errorf("could not create node from: %v of type %T and kind %v", v, v, v.Kind()))
+}
+
+// GetChild returns the child node at the given key, or nil if not found
+func (n *nodeimpl) GetChild(key string) node {
+	if n.typ != mapNodeType && n.typ != structNodeType {
+		return nil
+	}
+	if n.val.Kind() == reflect.Map {
+		inner := n.val.MapIndex(reflect.ValueOf(key))
+		if !inner.IsValid() {
+			return nil
+		}
+		if inner.IsNil() {
+			return nil
+		}
+		if inner.Kind() == reflect.Interface {
+			inner = inner.Elem()
+		}
+		return newNode(inner)
+	}
+	if n.val.Kind() == reflect.Struct {
+		findex := findFieldMatch(n.val, key)
+		if findex == -1 {
+			return nil
+		}
+		inner := n.val.Field(findex)
+		if !inner.IsValid() {
+			return nil
+		}
+		if inner.Kind() == reflect.Interface {
+			inner = inner.Elem()
+		}
+		return newNode(inner)
+	}
+	return nil
+}
+
+// ChildrenKeys returns the list of keys of the children of the given node
+func (n *nodeimpl) ChildrenKeys() []string {
+	if n.val.Kind() == reflect.Map {
+		keyvals := n.val.MapKeys()
+		keys := make([]string, 0, len(keyvals))
+		for _, kv := range keyvals {
+			kinf := kv.Interface()
+			kstr, ok := kinf.(string)
+			if ok {
+				keys = append(keys, kstr)
+			}
+		}
+		return keys
+	}
+	return nil
+}
+
+// AsScalar returns the node as a leaf if possible, otherwise nil
+func (n *nodeimpl) AsScalar() leaf {
+	if n.typ != scalarNodeType {
+		return nil
+	}
+	return &leafimpl{val: n.val}
+}
+
+// AsList returns the node as a list if possible, otherwise nil
+func (n *nodeimpl) AsList() nodeList {
+	if n.typ != listNodeType {
+		return nil
+	}
+	return &nodeListImpl{val: n.val}
+}
+
+// Size returns the number of elements in the list
+func (n *nodeListImpl) Size() int {
+	return n.val.Len()
+}
+
+// Index returns the kth element of the list
+func (n *nodeListImpl) Index(k int) node {
+	elem := n.val.Index(k)
+	if elem.Kind() == reflect.Interface {
+		elem = elem.Elem()
+	}
+	return newNode(elem)
+}
+
+// GetBool returns the scalar as a bool, or an error otherwise
+func (n *leafimpl) GetBool() (bool, error) {
+	if n.val.Kind() == reflect.Bool {
+		return n.val.Bool(), nil
+	}
+	if n.val.Kind() == reflect.String {
+		if n.val.String() == "true" {
+			return true, nil
+		} else if n.val.String() == "false" {
+			return false, nil
+		}
+	}
+	return false, conversionError(n.val, "bool")
+}
+
+// GetInt returns the scalar as a int, or an error otherwise
+func (n *leafimpl) GetInt() (int, error) {
+	switch n.val.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return int(n.val.Int()), nil
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return int(n.val.Uint()), nil
+	}
+	return 0, conversionError(n.val, "int")
+}
```

Comment:
YAML supports floats. Do we care about floats for our use case?

Reply:
We seem to use float values on our configuration template `pkg/config/config_template.yaml`

Reply:
Added float support to the various places that needed it.